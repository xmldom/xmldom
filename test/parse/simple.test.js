'use strict';

const { describe, expect, test } = require('@jest/globals');
const { MIME_TYPE, ParseError } = require('../../lib/conventions');
const { getTestParser } = require('../get-test-parser');

describe('parse', () => {
	test('simple', () => {
		const { errors, parser } = getTestParser();

		const actual = parser.parseFromString('<html><body title="1<2"></body></html>', 'text/html').toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});

	test('svg test', () => {
		const svgCase = [
			'<svg>',
			'  <metadata>...</metadata>',
			'  <defs id="defs14">',
			'  <path id="path4" d="M 68.589358,...-6.363961,-6.363964 z" />',
			'  <path id="path4" d="M 68.589358,...-6.363961,-6.363964 z" /></defs>',
			'</svg>',
		].join('\n');
		const { errors, parser } = getTestParser({ locator: {} });

		const actual = parser.parseFromString(svgCase, MIME_TYPE.XML_TEXT).toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});

	test('wrong closing tag', () => {
		const { errors, parser } = getTestParser({ locator: {} });

		const actual = parser
			.parseFromString(
				// TODO: xml not well formed but no warning or error, extract into different test?
				'<html><body title="1<2"><table>&lt;;test</table></body></html>',
				'text/html'
			)
			.toString();

		expect({ actual, ...(errors.length ? { errors } : undefined) }).toMatchSnapshot();
	});

	describe('invalid input', () => {
		test.each([
			['falsy string', ''],
			['object', {}],
			['number', 12345],
			['null', null],
		])('%s', (msg, testValue) => {
			const { parser } = getTestParser();

			expect(() => parser.parseFromString(testValue, MIME_TYPE.XML_TEXT)).toThrow(ParseError);
		});
	});
});
