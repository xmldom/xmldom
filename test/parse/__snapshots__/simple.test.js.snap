// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse line error 1`] = `
Object {
  "lineNumber": 7,
  "warning": Array [
    "[xmldom warning]	unclosed xml attribute
@#[line:7,col:3]",
  ],
}
`;

exports[`parse simple 1`] = `
Object {
  "actual": "<html xmlns=\\"http://www.w3.org/1999/xhtml\\"><body title=\\"1&lt;2\\"></body></html>",
}
`;

exports[`parse svg test 1`] = `
Object {
  "actual": "<svg>
  <metadata>...</metadata>
  <defs id=\\"defs14\\">
  <path id=\\"path4\\" d=\\"M 68.589358,...-6.363961,-6.363964 z\\"/>
  <path id=\\"path4\\" d=\\"M 68.589358,...-6.363961,-6.363964 z\\"/></defs>
</svg>",
}
`;

exports[`parse unclosed inner 1`] = `
Object {
  "actual": "<r><Page><Label/></Page>  <Page/></r>",
  "error": Array [
    "[xmldom error]	end tag name: Page maybe not complete
@#[line:1,col:10]",
  ],
}
`;

exports[`parse unclosed root 1`] = `
Object {
  "actual": "<Page><Label class=\\"title\\"/></Page>  1",
  "error": Array [
    "[xmldom error]	end tag name: Page is not complete:Page
@#[line:1,col:20]",
  ],
}
`;

exports[`parse wrong closing tag 1`] = `
Object {
  "actual": "<html xmlns=\\"http://www.w3.org/1999/xhtml\\"><body title=\\"1&lt;2\\"><table></table>&lt;;test</body></html>",
}
`;
